azure/secrets.go

```diff
@@ -0,0 +1,67 @@
+package azure
```

Comment:
As described in conversation, let's add an interface for cloud and implementations in a single package cloud.